lib/sidekiq/batch.rb

```diff
@@ -169,7 +169,7 @@ def parent
     end
 
     def valid?(batch = self)
-      valid = !Sidekiq.redis { |r| r.exists("invalidated-bid-#{batch.bid}") }
+      valid = Sidekiq.redis { |r| r.exists("invalidated-bid-#{batch.bid}") }.zero?
```

Comment:
According to the documentation of [EXISTS](https://redis.io/commands/exists/) in redis, this function will return an integer.